common/metrics/metrics.go

```diff
@@ -31,21 +30,30 @@ var dashboardClient *dashboard.DashboardsClient
 
 // MonitoringMetricsResources contains information required to create the monitoring dashboard
 type MonitoringMetricsResources struct {
-	ProjectId            string
-	DataflowJobId        string
-	DatastreamId         string
-	GcsBucketId          string
-	PubsubSubscriptionId string
-	SpannerInstanceId    string
-	SpannerDatabaseId    string
-	ShardId              string
+	ProjectId                string
+	DataflowJobId            string
+	DatastreamId             string
+	GcsBucketId              string
+	PubsubSubscriptionId     string
+	SpannerInstanceId        string
+	SpannerDatabaseId        string
+	ShardToDataStreamNameMap map[string]string
+	ShardToDataflowInfoMap   map[string]internal.ShardedDataflowJobResources
+	ShardToPubsubIdMap       map[string]internal.PubsubCfg
+	ShardToGcsMap            map[string]internal.GcsResources
+	ShardId                  string
 }
```

Comment:
I think it probably makes sense have structs in separate file, utility functions in a separate file, and shard-level and aggregated dashboard creation stuff in their own files. 

`metrics.go` is already a long file and this will soon become hard to manage if we add more metrics in the future.